### PR TITLE
Feature/configure additional OIDC scopes

### PR DESCRIPTION
**Pull request by tvanderwal-lmco**
Implements the ability to request additional scopes from the OIDC provider documented in issue [4349](https://github.com/treeverse/lakeFS/issues/4349)

### CODE REVIEW

**Comment by tvanderwal-lmco** [comment]
Outside of the unit tests I did confirm functionality locally by requesting the `email` scope and using it as the friendly name. 

```
---
listen_address: "0.0.0.0:8000"

database:
  type: "local"
  local:
    path: "~/lakefs/metadata"

logging:
  format: text
  level: DEBUG
  output: "-"

auth:
  encrypt:
    secret_key: "example"
  oidc:
    enabled: true
    client_id: myclient
    client_secret: clientsecret
    callback_base_url: http://localhost:8000
    url: https://oidcprovider
    default_initial_groups: ["Admins"]
    is_default_login: true
    friendly_name_claim_name: email
    additional_scope_claims: 
      - upn
      - email

blockstore:
  type: local
  local:
    path: "~/lakefs/dev/data"

gateways:
  s3:
    region: us-east-1
```
Screen Shot 2022-10-31 at 11 16 22 AM

**Comment by arielshaqed** [comment]
Thanks!
Running tests and putting some labels on it.

File: docs/reference/configuration.md [hunk 1]
```diff
@@ -96,10 +96,11 @@ This reference uses `.` to denote the nesting of values.
 * `auth.oidc.url` `(string : )` - The base URL of your OIDC compatible identity provider.
 * `auth.oidc.callback_base_url` `(string : )` - The scheme, host and port of your lakeFS installation. After authenticating, your identity provider will redirect you to a URL under this base.
 * `auth.oidc.default_initial_groups` `(string[] : [])` - By default, OIDC users will be assigned to these groups
-* `auth.oidc.initial_groups_claim_name` `(string[] : [])` - Use this claim from the ID token to provide the initial group for new users.
+* `auth.oidc.initial_groups_claim_name` `(string[] : [])` - Use this claim from the ID token to provide the initial group for new users. This will take priority if `auth.oidc.default_initial_groups` is also set.
```
Comment (guy-har) [inline docs/reference/configuration.md:99]: 🙏

**Review by guy-har — APPROVED** [review]
LGTM
Love the tests 
Thank you for this contribution!!

**Pull request merged**